### PR TITLE
Add camel-saxon integration test

### LINKED ISSUE #352: Add camel-saxon integration tests
camel-saxon integration tests

### PR DESCRIPTION
**Pull request by f2par0**
fixes #352  camel-saxon integration test

### CODE REVIEW

File: tests/features/camel-saxon/src/main/java/org/apache/karaf/camel/test/CamelSaxonRouteSupplier.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        name = "karaf-camel-saxon-test",
+        immediate = true,
+        service = CamelRouteSupplier.class
+)
+public class CamelSaxonRouteSupplier extends AbstractCamelSingleFeatureResultMockBasedRouteSupplier {
+
+
+    @Override
+    protected boolean consumerEnabled() {
+        return  false;
```
Comment (essobedo) [inline tests/features/camel-saxon/src/main/java/org/apache/karaf/camel/test/CamelSaxonRouteSupplier.java:34]:
```suggestion
        return false;
```

File: tests/features/camel-saxon/src/main/java/org/apache/karaf/camel/test/CamelSaxonRouteSupplier.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    protected boolean consumerEnabled() {
+        return false;
+    }
+
+    @Override
+    protected void configureProducer(RouteBuilder builder, RouteDefinition producerRoute) {
+        producerRoute.log("calling xquery endpoint")
+                .setBody(builder.constant("<person><city>London</city></person>"))
+                .choice()
+                .when().xpath("person/city = 'London'")
```
Comment (essobedo) [inline tests/features/camel-saxon/src/main/java/org/apache/karaf/camel/test/CamelSaxonRouteSupplier.java:42]: Why don't you use an XQuery predicate?

Comment (f2par0) [reply tests/features/camel-saxon/src/main/java/org/apache/karaf/camel/test/CamelSaxonRouteSupplier.java]: Fixed, it was a copy of the test in the camel project.

**Review by essobedo — APPROVED** [review]
LGTM

**Pull request merged**